packages/discord.js/src/structures/Message.js

```diff
@@ -780,12 +780,16 @@ class Message extends Base {
    * @readonly
    */
   get pinnable() {
-    const { channel } = this;
-    return Boolean(
-      !this.system &&
-        (!this.guild ||
-          (channel?.viewable &&
-            channel?.permissionsFor(this.client.user)?.has(PermissionFlagsBits.ManageMessages, false))),
+    if (this.system || !this.channel?.viewable) return false;
+
+    const permissions = this.channel.permissionsFor(this.client.user);
+    if (!permissions) return false;
+
+    return (
+      !this.guild ||
+      permissions.has([PermissionFlagsBits.ReadMessageHistory, PermissionFlagsBits.PinMessages], false) ||
+      (!this.guild.features.includes('PIN_PERMISSION_MIGRATION_COMPLETE') &&
```

Comment:
As of 0ff239a602aed71c74c931c975c84900cf671312, we can use `GuildFeature.PinPermissionMigrationComplete`!